types/addons-linter/index.d.ts

```diff
@@ -0,0 +1,143 @@
+         * The log-level to generate.
+         */
+        logLevel: "fatal" | "error" | "warn" | "info" | "debug" | "trace";
+
+        /**
+         * Treat warnings as errors.
+         */
+        warningsAsErrors?: boolean;
+
+        /**
+         * The type of output to generate.
+         */
+        output?: "json" | "text" | "none";
+
+        /**
+         * Output only metadata as JSON.
+         */
+        metadata?: boolean;
+
+        /**
+         * Prettify JSON output.
+         */
+        pretty?: boolean;
+
+        /**
+         * Show stacktraces when errors are thrown.
+         */
+        stack?: boolean;
+
+        /**
+         * Disable colorful shell output.
+         */
+        boring?: boolean;
+
+        /**
+         * Treat the input file (or directory) as a privileged extension.
+         */
+        privileged?: boolean;
+
+        /**
+         * Disable messages related to hosting on addons.mozilla.org.
+         */
+        selfHosted?: boolean;
+
+        /**
+         * Enable MV3 background service worker support.
+         */
+        enableBackgroundServiceWorker?: boolean;
+
+        /**
+         * Set a custom minimum allowed value for the manifest_version property
+         */
+        minManifestVersion?: number;
+
+        /**
+         * Set a custom maximum allowed value for the manifest_version property
+         */
+        maxManifestVersion?: number;
+
+        /**
+         * Disable list of comma separated eslint rules.
+         */
+        disableLinterRules?: string;
+
+        /**
+         * Disable the auto-close feature when linting XPI files.
+         */
+        disableXpiAutoclose?: boolean;
+
+        /**
+         * Exclude files
+         */
+        shouldScanFile?: (fileOrDirName: string, isDir: boolean) => boolean;
+
+        /**
+         * Scan a selected file.
+         */
+        scanFile?: string[];
+    };
+
+    /**
+     * This prevent the linter to exit the nodejs application.
+     */
+    runAsBinary?: boolean;
+}
+
+export interface LinterResults {
+    count: number;
+
+    summary: {
+        errors: number;
+        notices: number;
+        warnings: number;
+    };
+
+    metadata: Record<string, unknown>;
+
+    scanFile: string[];
+
+    errors: Array<{
+        _type: "error";
+        code: string;
+        message: string;
+        description: string;
+        file: string;
+    }>;
+
+    notices: Array<{
+        _type: "notice";
+        code: string;
+        message: string;
+        description: string;
+        file: string;
+    }>;
+
+    warnings: Array<{
+        _type: "warning";
+        code: string;
+        message: string;
+        description: string;
+        file: string;
+    }>;
+}
+
+export interface Linter {
+    run: () => Promise<LinterResults>;
+}
+
+export function createInstance(options: Options): Linter;
+
+declare const _default: {
+    createInstance: typeof createInstance;
+};
+
+export default _default;
```

Comment:
I don't think this is right; the upstream library is CJS and its exported object has this shape:

```console
$ npm i "addons-linter"

added 252 packages, and audited 253 packages in 4s

found 0 vulnerabilities
$ node -e 'console.log(require("addons-linter"))'
{
  Linter: [class Linter],
  createInstance: [Function: createInstance],
  isRunFromCLI: [Function: isRunFromCLI]
}
```

So there doesn't seem to be a default.

Reply:
It was my lack of confirmation.  Deleted default.

As a side note, as I added in the code comment, the Linter class and isRunFromCLI() function do not seem to be intended for external use, so I have not written a type definition for them.

Reply:
Yes, omitting stuff is totally fine.